syntax = "proto3";

message Register {
  string Password = 1;
  string GUID = 2;
  string Username = 3;
  string Hostname = 4;
}

message TaskRequest {
  string TaskGuid = 1;
  int32 Opcode = 2;
  string Args = 3;
}

message TaskResponse {
  string TaskGuid = 1;
  string GUID = 2; // implant ID (TODO need to change this for final project)
  bytes Response = 3;
}
